tests/src/Kernel/Validators/ValidatorValidDelimitedFileProcessTest.php

```diff
@@ -0,0 +1,561 @@
+      );
+
+      $current_row_index = 0;
+      // Loop through expectations for each row of a table.
+      foreach ($table as $expected_line_no => $expected_values) {
+        if ($expected_line_no == 'expected_message') {
+          continue;
+        }
+        // Select our current row as an array.
+        $select_row_cells = (array) $selected_rows[$current_row_index]->td;
+        // 1st Column: Line Number
+        $line_number = $select_row_cells[0];
+        $this->assertEquals($expected_line_no, $line_number, "Did not get the expected line number in the rendered table from processing ValidDelimitedFile failures.");
+        // 2nd Column: The contents of this line in the file
+        $line_contents = (string) $select_row_cells[1];
+        $this->assertEquals($expected_values['line_contents'], $line_contents, 'Did not get the expected line contents in the rendered table from processing ValidDelimitedFile failures.');
+        // Move onto the next row.
+        $current_row_index++;
+      }
+    }
+  }
+
+  /**
+   * Data Provider for triggering exceptions in process failures method.
+   *
+   * @return array
+   *   Each scenario is an array with the following:
+   *   - An array of validation status arrays that get passed to the process
+   *     method. It is keyed by the line number that triggered this failed
+   *     validation status, further keyed by:
+   *     - The line number that triggered this failed validation status.
+   *       - 'case': a developer-focused string describing the case checked.
+   *       - 'valid': FALSE to indicate that validation failed.
+   *       - 'failedItems': array of items that failed with the following keys:
+   *         - 'raw_row': The contents of the raw row as it appears in the file.
+   *         - 'expected_columns': The number of columns expected in the input
+   *           file as determined by calling getExpectedColumns().
+   *         - 'strict': A boolean indicating whether the number of expected
+   *           columns by the validator is strict (TRUE) or is the minimum
+   *           number required (FALSE).
+   *   - An array of tokens to use for altering the messages that get displayed
+   *     to the user. The key is the token, (ex. 'project'), and the value is
+   *     the new value to be shown for that token.
+   *   - An array of expectations that we want to find in the resulting rendered
+   *     output. This array is nested by the tables expected (keyed by type -
+   *     'unsupported' or 'delimited'), in the order they are expected to show
+   *     up on the page (1 array per table). Each array has the following keys:
+   *     - 'expected_message': The message expected in the return value of the
+   *       process method for this scenario.
+   *     - 1+ arrays keyed by the line number in the input file that triggered
+   *       the failed validation status, further keyed by:
+   *       - 'line_contents': The raw contents of this line that failed.
```

Comment:
This doesn't match up with what is in the data provider. How about:
```suggestion
   *   - An array of expectations that we want to find in the resulting rendered
   *     output. It has the following keys:
   *     - 'expected_message': The message expected in the return value of the
   *       process method for this scenario.
```